src/components/AdvancedSearchFilterInput.tsx

```diff
@@ -25,6 +25,14 @@ export default function AdvancedSearchFilterInput({
 
   const handleKeyChange = (value) => {
     setFilterKey(value);
+    const selected = fields.find((field) => field.value === value);
+
+    // Set the value to either the default option, or blank when changing the filter type
+    if (selected.options && selected.options.length) {
+      setFilterValue(selected.options[0]);
+    } else {
+      setFilterValue("");
+    }
```

Comment:
There could be a problem here. This PR is introducing the potential for a reduced set of operators, so we'll also need to ensure that we've got a valid operator for the newly current field. Ideally, we'd only change the `filterOp` if the current one is not supported.

It would be useful to add a test simulating going from an op on a field with broader set ops to a field that doesn't support that op, then adding the filter, and finally verifying that the key, value, and op are as expected.

Reply:
Good catch! Thank you, totally missed this one.